File: src/Application Specific/Classes/Interfaces/IUserControl.vb
Public Interface IUserControl

    Sub LoadForm(ByVal sender As Object, ByVal e As System.EventArgs)

    ReadOnly Property LoadedItem() As Object

    Event RecordsChanged(ByVal dv As DataView, ByVal pageIn As Entity.Sys.Enums.PageViewing, ByVal FromASave As Boolean, ByVal FromADelete As Boolean, ByVal extraText As String)

    Event StateChanged(ByVal newID As Integer)

    Sub Populate(Optional ByVal ID As Integer = 0)

    Function Save() As Boolean


End Interface

File: src/Application Specific/Classes/Interfaces/IUserControl.cs
using System;
using System.Data;

namespace FSM
{
    public interface IUserControl
    {
        void LoadForm(object sender, EventArgs e);

        object LoadedItem { get; }

        event RecordsChangedEventHandler RecordsChanged;

        delegate void RecordsChangedEventHandler(DataView dv, Entity.Sys.Enums.PageViewing pageIn, bool FromASave, bool FromADelete, string extraText);

        event StateChangedEventHandler StateChanged;

        delegate void StateChangedEventHandler(int newID);

        void Populate(int ID = 0);
        bool Save();
    }
}